subsys/bluetooth/enocean.c

```diff
@@ -217,6 +220,26 @@ static void handle_switch_data(const struct bt_le_scan_recv_info *info,
 	cb->button(dev, action, status >> 1, opt_data, opt_data_len);
 }
 
+static void handle_switch_payload(const struct bt_le_scan_recv_info *info,
+				  struct net_buf_simple *buf, const uint8_t *payload, uint8_t len,
+				  bool has_shortened_name)
+{
+	/* The data format is decided by a mix of lengths and bitfields */
+	if (len == 29 || (has_shortened_name && len == 23)) {
+		if (commissioning) {
+			handle_switch_commissioning(info, buf, has_shortened_name);
+		}
+		return;
+	}
+
+	if (!has_shortened_name && !(payload[8] & (BIT_MASK(3) << 5)) && len >= 12 && len <= 16) {
```

Comment:
Nitpick: Could the `!(payload[8] & (BIT_MASK(3) << 5))` check be moved inside `handle_switch_data`? I feel it might make it a bit cleaner

Reply:
No, I need it to proceed to the "LOG_DBG" in case these any of these bits of the payload are set (because that could in the future be a valid message for a new switch firmware that we don't support, and then it should log it so we (or customers) notice, whereas putting the check inside `handle_switch_data` would silently ignore it)

Reply:
Basically, this if is supposed to be `if (we are sure this is a switch data packet) {`, and then it needs this bit check.